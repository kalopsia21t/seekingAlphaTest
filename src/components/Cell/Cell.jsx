import React from 'react';
import classnames from 'classnames';
import styles from './Cell.less'

class Cell extends React.PureComponent {
    static defaultProps = {
        color: false
    };
    
    constructor (props) {
        super(props);
    }
    
    render() {
        const { color } = this.props;
        return (<div className={classnames(styles.cell, color ? styles.cellBlack : styles.cellWhite )}>

        </div>)
    }
}

export default Cell;