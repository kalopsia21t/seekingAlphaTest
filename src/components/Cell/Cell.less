.cell {
    display: block;
    width: 10px;
    height: 10px;
}

.cellWhite {
    background: white;
}

.cellBlack {
    background: black;
}