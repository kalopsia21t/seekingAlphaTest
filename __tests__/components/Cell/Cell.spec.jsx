import React from 'react';
import { shallow, mount } from 'enzyme';

import Cell from '../../../src/components/Cell/Cell';

describe('<Cell />', () => {
    // test('must render black if color prop is true', () => {
    //     const container = mount(<Cell color={true}/>);
    //     expect(container).toMatchSnapshot();
    // });

    test('must render white if color prop is false', () => {
        const container = mount(<Cell color={false}/>);
        expect(container).toMatchSnapshot();
    });

    test('must have property color equal false if props wasn\'t passed', () => {
        const container = mount(<Cell />);
        expect(container).toHaveProp({ color: false });
    });
});