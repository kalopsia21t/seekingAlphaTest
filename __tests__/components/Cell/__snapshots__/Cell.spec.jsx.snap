// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> must render white if color prop is false 1`] = `
<Cell
  color={false}
>
  <div
    className="cell cellWhite"
  />
</Cell>
`;
